viewing_party/party.py

```diff
@@ -0,0 +1,184 @@
+
+def get_watched_avg_rating(user_data):
+  user_rating_total = 0.0
+  average = 0.0
+  watched_list = user_data["watched"]
+  
+  if watched_list != []:
+    for movie_dict in watched_list:
+      user_rating_total += movie_dict["rating"] 
+    average = user_rating_total / len(watched_list)
+    
+  return average
+
+      
+def get_most_watched_genre(user_data):
+  popular_genre_list = []
+  count = 0
+  watched_list = user_data["watched"]
+  popular_genre = None
+  
+  if watched_list != []:
+    for movie_dict in watched_list:
+      popular_genre_list.append(movie_dict["genre"])
+      
+    for genre_name in popular_genre_list:
+      
+      if popular_genre_list.count(genre_name) > count:
+        count = popular_genre_list.count(genre_name)
+        popular_genre = genre_name
+  
+  return popular_genre
+
+
+# Wave 3
+
+def get_unique_watched(user_data):
+    
+  # Movies the user has watched, but none of their friends have watched
+  
+  friends_list = user_data
+  user_movies_list = []
+  friends_movies_list = []
+  user_unique_movies = []
+  friends_unique_movies = {}
+  friends_movie_dict_list = user_data["friends"]
+
+
+  for friends_dict in friends_movie_dict_list:
+    for movie_watched_dict in friends_dict["watched"]:
+        friends_movies_list.append(movie_watched_dict["title"])
+
+  for movie_dict in user_data["watched"]:
+    if movie_dict["title"] not in friends_movies_list:
+      user_unique_movies.append(movie_dict)
+      
+  return user_unique_movies
+
+
+def get_friends_unique_watched(user_data):
+    
+  # Movies at least one of the user's friends have watched, but the user has not watched
+    
+  friends_list = user_data
+  user_movies_list = []
+  friends_movies_list = []
+  user_unique_movies = []
+  friends_unique_movies = []
+  the_friends_with_movie_dict_list = user_data["friends"]
+  
+  for movie_dict in user_data["watched"]:
+      user_movies_list.append(movie_dict["title"])
+
+  for friends_dict in the_friends_with_movie_dict_list:
+      for movie_watched_dict in friends_dict["watched"]:
+          if movie_watched_dict["title"] not in user_movies_list:
+              if movie_watched_dict not in friends_unique_movies:
+                  friends_unique_movies.append(movie_watched_dict)
+  
+  return friends_unique_movies
+
+
+# Wave 4
+
+def get_available_recs(user_data):
+    
+    host_list = []
+    recommendation_list = []
+    subscription_list = user_data["subscriptions"]
+    user_not_watched_list = get_friends_unique_watched(user_data)
+    
+    for movie_dict in user_not_watched_list:
+        if movie_dict["host"] in subscription_list:
+            recommendation_list.append(movie_dict)
+    
+    return recommendation_list
+
+
+# Wave 5 
+
+def get_new_rec_by_genre(user_data):
+  
+  user_fave_genre = get_most_watched_genre(user_data)
+  friends_watched_list = get_friends_unique_watched(user_data)
+  new_rec_list = []
+  
+  if len(friends_watched_list) == 0:
+    return new_rec_list
```

Comment:
This is a nice guard clause (does a quick check and bails out of the function so we don't need to worry about it during the rest of the function). But Notice that even without this check, if `friends_watched_list` was empty, the retuned value would still be an empty list.

Also, a more pythonic way to check for an empty list is
```python
if not friends_watched_list:
```